pkg/resource/function/hooks.go

```diff
@@ -0,0 +1,348 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package function
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+// customUpdateFunction patches each of the resource properties in the backend AWS
+// service API and returns a new resource with updated fields.
+func (rm *resourceManager) customUpdateFunction(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateFunction")
+	defer exit(err)
+
+	if delta.DifferentAt("Spec.Code") {
+		err = rm.updateFunctionCode(ctx, desired)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if delta.DifferentAt("Spec.Tags") {
+		err = rm.updateFunctionTags(ctx, latest, desired)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if delta.DifferentAt("Spec.Description") ||
+		delta.DifferentAt("Spec.Handler") ||
+		delta.DifferentAt("Spec.KMSKeyARN") ||
+		delta.DifferentAt("Spec.MemorySize") ||
+		delta.DifferentAt("Spec.Role") ||
+		delta.DifferentAt("Spec.Timeout") ||
+		delta.DifferentAt("Spec.Environement") ||
+		delta.DifferentAt("Spec.FileSystemConfigs") ||
+		delta.DifferentAt("Spec.ImageConfig") ||
+		delta.DifferentAt("Spec.TracingConfig") ||
+		delta.DifferentAt("Spec.VPCConfig") {
+		err = rm.updateFunctionConfiguration(ctx, desired)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	readOneLatest, err := rm.ReadOne(ctx, desired)
```

Comment:
nit: Worth adding a comment here explaining why is this ReadOne call needed.

Reply:
Will add that in the next PR, adding `Concurrency` field! Thanks for the help!